skbio/sequence/_sequence.py

```diff
@@ -187,30 +351,107 @@ def metadata(self):
     def metadata(self, metadata):
         if not isinstance(metadata, dict):
             raise TypeError("metadata must be a dict")
-        self._metadata = metadata
+        # shallow copy
+        self._metadata = metadata.copy()
+
+    @metadata.deleter
+    def metadata(self):
+        if hasattr(self, '_metadata'):
+            del self._metadata
+        self.metadata = {}
 
     @property
     def positional_metadata(self):
-        """Pandas DataFrame containing metadata on a per-character basis.
+        """pandas.DataFrame containing metadata on a per-character basis.
```

Comment:
:+1: I'll go with `pd.DataFrame` since we're generally using `np.something` in the sequence docs